authzed/api/v1/experimental_service.proto

```diff
@@ -136,3 +184,169 @@ message BulkExportRelationshipsResponse {
+
+message ExperimentalReflectSchemaRequest {
+  Consistency consistency = 1;
+ 
+  // optional_filters defines optional filters that are applied in
+  // an OR fashion to the schema, before being returned
+  repeated ExpSchemaFilter optional_filters = 2;
+}
+
+message ExperimentalReflectSchemaResponse {
+  // definitions are the definitions defined in the schema.
+  repeated ExpDefinition definitions = 1;
+
+  // caveats are the caveats defined in the schema.
+  repeated ExpCaveat caveats = 2;
+
+  // read_at is the ZedToken at which the schema was read.
+  ZedToken read_at = 3;
+}
+
+message ExpSchemaFilter {
+  enum KindFilter {
+    KIND_FILTER_UNSPECIFIED = 0;
+    KIND_FILTER_DEFINITION = 1;
+    KIND_FILTER_CAVEAT = 2;
+    KIND_FILTER_RELATION = 3;
+    KIND_FILTER_PERMISSION = 4;
+  }
+
+  // optional_definition_name_match is a regex that is matched against the definition or caveat name.
+  // If not specified, will be ignored.
+  string optional_definition_name_match = 1;
+
+  // optional_relation_or_permission_name_match is a regex that is matched against the relation or permission name.
+  // If not specified, will be ignored.
+  string optional_relation_or_permission_name_match = 2;
+
+  // kind_filters is a list of kinds to filter on. If not specified, will be ignored. If multiple are specified,
+  // the filter will be applied in an OR fashion.
+  repeated KindFilter kind_filters = 3;
+}
+
+message ExpDefinition {
+  string name = 1;
+  string comment = 2;
+  
+  repeated ExpRelation relations = 3;
+  repeated ExpPermission permissions = 4;
+}
+
+message ExpCaveat {
+  string name = 1;
+  string comment = 2;
+
+  repeated ExpCaveatParameter parameters = 3;
+  string expression = 4;
+}
+
+message ExpCaveatParameter {
+  string name = 1;
+  string type = 2;
+  string parent_caveat_name = 3;
+}
+
+message ExpRelation {
+  string name = 1;
+  string comment = 2;
+  string parent_definition_name = 3;
+  repeated ExpTypeReference subject_types = 4;
+}
+
+message ExpTypeReference {
+  // subject_definition_name is the name of the subject's definition.
+  string subject_definition_name = 1;
+
+  // optional_caveat_name is the name of the caveat that is applied to the subject, if any.
+  string optional_caveat_name = 2;
+
+  oneof typeref {
+    // is_terminal_subject is true if the subject is terminal, meaning it is referenced directly vs a sub-relation.
+    bool is_terminal_subject = 3;
+    
+    // optional_relation_name is the name of the relation that is applied to the subject, if any.
+    string optional_relation_name = 4;
+
+    // is_public_wildcard is true if the subject is a public wildcard.
+    bool is_public_wildcard  = 5;
+  }
+}
+
+message ExpPermission {
+  string name = 1;
+  string comment = 2;
+  string parent_definition_name = 3;
+}
+
+message ExperimentalComputablePermissionsRequest {
+  Consistency consistency = 1;
+  repeated ExpRelationReference relations = 3;
```

Comment:
what about including a filter so that you can avoid objects you're not interested in? i.e. when looking up the effects of a relationship that is used in an arrow

Reply:
A filter for the response you mean?

Reply:
Yeah - you may be querying for changes to `folder#parent` and want to ignore changes in `document#view` for example

Reply:
Added